_pytest/cacheprovider.py

```diff
@@ -147,11 +142,11 @@ def pytest_collection_modifyitems(self, session, config, items):
                     previously_failed.append(item)
                 else:
                     previously_passed.append(item)
-            if not previously_failed and previously_passed:
```

Comment:
I didn't quite understand why `previously_passed` was being considered here for this condition... to me it makes sense to skip deselecting items if we didn't find any previous failures in the current collection.